File: Python/ValidPalindrome.py
import re

def is_valid_palindrome(s):
    s = s.lower()
    s = re.sub(r'[^a-z0-9]', '', s)   # remove non-alphanumeric characters
    n = len(s)
    
    for i in range(n // 2):
        if s[i] != s[n - 1 - i]:
            return False
    return True


user_input = input("Enter a string: ")
if is_valid_palindrome(user_input):
    print("The string is a valid palindrome.")
else:
    print("The string is not a valid palindrome.")

File: Java/Practice/ValidPalindrome.java
package Practice;

import java.util.Scanner;
    //valid palindrome function class 
public class ValidPalindrome {
    public boolean Valid(String s) {
      Scanner sc =new Scanner (System.in);
      System.out.print("Enter the String :");
      s = sc.nextLine();
      s=s.toLowerCase();  //converting to lower case
      s=s.replaceAll("[^a-z0-9]","");// removing all non alphanumeric characters
      sc.close();
      for(int i=0;i<s.length()/2;i++){
          if(s.charAt(i)!=s.charAt(s.length()-1-i)){
              return false;
          }
      }
     
      return true;
    }

    //main class
    public static void main(String args[]){
        ValidPalindrome obj=new ValidPalindrome(); //creating object of the class
        String s="";
        if(obj.Valid(s)){
            System.out.println("The given string is a palindrome");
        }
        else{
            System.out.println("The given string is not a palindrome");
        }
    }
}
